### DISCUSSION TITLE
pycuda.driver.CompileError: nvcc compilation of /tmp/tmp7hnwb4te/kernel.cu failed

### DISCUSSION THREAD

**Discussion by tiandao091** [inducer/pycuda#422] [Q&A]
python version: 3.9
cuda version: 11.6
pycuda version: pip install pycuda=2022.1

i test my first cuda code in python, but failed. The error information is here:
Traceback (most recent call last):
  File "/home/hdb/project/DIP_db/test_pycuda/test_pycuda.py", line 30, in <module>
    mod = SourceModule(kernel_code)
  File "/opt/anaconda3/envs/pytorch/lib/python3.9/site-packages/pycuda/compiler.py", line 352, in __init__
    cubin = compile(
  File "/opt/anaconda3/envs/pytorch/lib/python3.9/site-packages/pycuda/compiler.py", line 301, in compile
    return compile_plain(source, options, keep, nvcc, cache_dir, target)
  File "/opt/anaconda3/envs/pytorch/lib/python3.9/site-packages/pycuda/compiler.py", line 154, in compile_plain
    raise CompileError(
_pycuda.driver.CompileError: nvcc compilation of /tmp/tmp7hnwb4te/kernel.cu failed_
[command: nvcc --cubin -arch sm_80 -I/opt/anaconda3/envs/pytorch/lib/python3.9/site-packages/pycuda/cuda kernel.cu]
[stderr:
_kernel.cu(6): error: expression must be a modifiable lvalue_

_1 error detected in the compilation of "kernel.cu"._

I don't know why. So can you give me some advices?

**Comment by inducer** [comment]
Looks like you've got a C/C++ error in your kernel code:
```
kernel.cu(6): error: expression must be a modifiable lvalue
```

**Comment by tiandao091** [comment]
Thank you for your information, but i can't find the kernel.cu file in this path "opt/anaconda3/envs/pytorch/lib/python3.9/site-packages/pycuda/cuda kernel.cu".
![image](https://github.com/inducer/pycuda/assets/3324257/90b3b583-d012-410a-a795-acc67d001df1)
